File: TypeScript/RecursionExtra/sumArray.ts
// Sum of array elements
//
// No loops
//
// Example: [1, 2, 3, 4] → 10

function sumArray(arr: number[], i: number): number {
    if (i >= arr.length) {
        return 0;
    }
    return arr[i] + sumArray(arr, i + 1);
}

File: Golang/RecursionExtra/sumArray.go
// Sum of array elements
//
// No loops
//
// Example: [1, 2, 3, 4] → 10

package main

func sumPtr(arr *[]int, i int) int {
	if i >= len(*arr) {
		return 0
	}
	return (*arr)[i] + sumPtr(arr, i+1)
}
